### ISSUE TITLE
Frameworks/App.framework/App does not contain that architecture

### ISSUE THREAD

**Issue by karandaid** [flutter/flutter#22749]
I can't execute my flutter app over ios simulator

```
## Launching lib/main.dart on iPhone X in debug mode...
Running pod install...
Starting Xcode build...
Xcode build done.
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
    
Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    ld: warning: ignoring file <url>/ios/Flutter/App.framework/App, missing required architecture x86_64 in file <url>os/Flutter/App.framework/App (2 slices)
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    fatal error: lipo: -extract x86_64 specified but fat file: <url>/ios/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App does not contain that architecture
    Failed to extract x86_64 for <url>/ios/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App. Running lipo -info:
    Architectures in the fat file: <url>/ios/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App are: armv7 arm64 
Could not build the application for the simulator.
Error launching application on iPhone X.
```

## Logs

```
flutter doctor -v
[✓] Flutter (Channel beta, v0.7.3, on Mac OS X 10.13.4 17E199, locale en-CA)
    • Flutter version 0.7.3 at /Applications/flutter
    • Framework revision 3b309bda07 (5 weeks ago), 2018-08-28 12:39:24 -0700
    • Engine revision af42b6dc95
    • Dart version 2.1.0-dev.1.0.flutter-ccb16f7282

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.0)
    • Android SDK at /Users/karandeepsinghdhaliwal/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.4, Build version 9F1027a
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.4155.35

[!] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (1 available)
    • iPhone X • 42BE8DC6-84B6-481A-81D4-AA400CD50848 • ios • iOS 11.4 (simulator)

! Doctor found issues in 1 category.

```

**Comment by zoechi** [comment]
Can you reproduce with a new project (created with `flutter create foo`)?

**Comment by zoechi** [comment]
@satyasreem 
Please add the output of `flutter doctor -v`.

**Comment by satyasreem** [comment]
```
$ flutter doctor --verbose
[✓] Flutter (Channel beta, v0.9.4, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.9.4 at <Flutterhome>/flutter
    • Framework revision f37c235c32 (5 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at <UserPath>/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.2
    • Dart plugin version 181.5616
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.28.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.19.0

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!`
```

**Comment by satyasreem** [comment]
Is there any resolution for this issue?

**Comment by satyasreem** [comment]
This can be closed. Creating a new project and copying over iOS files fixed this issue. ( See my comment on https://github.com/flutter/flutter/issues/24479)

**Comment by ChekTek** [comment]
I resolved this issue by changing the flutter target runner in xcode -  build active architecture only: yes
![image](https://user-images.githubusercontent.com/6678919/53133126-0899ff80-3540-11e9-9339-584a73afcbdf.png)
then run flutter clean and flutter run! :)

**Comment by suesitran** [comment]
changing Build Active Architecture Only does not help me

**Comment by SameerDonga** [comment]
@suesitran did you find any solution?

**Comment by elliott-io** [comment]
None of this worked for me. In Xcode, I noticed that the Pods project Build Settings has Base SDK set to No SDK (macOS) and Valid Architectures to i386 and x86_64. When I change the Base SDK to iOS, the Valid Architectures change, but running `flutter build ios` resets this and the command succeeds. However, when I try to run the project in Xcode on the iPhone 11 Pro simulator, the build fails and I get this error:

`fatal error: lipo: -extract x86_64 specified but fat file: /Users/username/Library/Developer/Xcode/DerivedData/Runner-frvesgtnbskrszfesjbgjiqntvyv/Build/Products/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App does not contain that architecture`

If I try to run from VS Code gives the same error.

I am using several plugins and I noticed that these three reference x86_64 architecture for iOS simulator (removing this just gets reset when doing flutter build ios):

  google_sign_in: ^4.0.14
  path_provider: ^1.4.2
  shared_preferences: ^0.5.4+3

I was on the dev channel v1.12.2. I switched to the master channel with hopes there was a fix.
Flutter 1.12.5-pre.24 • channel master • https://github.com/flutter/flutter.git
Framework • revision 6e5769d425 (3 hours ago) • 2019-11-19 11:19:44 -0800
Engine • revision ecb32952ff
Tools • Dart 2.7.0

Development has been completely stopped so any guidance would be much appreciated.

**Comment by biscoe916** [comment]
This should be re-opened.

**Comment by DhavalRKansara** [comment]
> This should be re-opened.

Yeah, I am also facing the same issues in my Application.

Once I have run my build in the real device then I am not able to run it into any simulator and vice versa. I have already deleted derived data multiple times and tried to run but still getting the same error.
```
Non-fat binary /Users/dhavalkansara/Library/Developer/Xcode/DerivedData/Runner-dwtuugielmxutoatpwezjbbcemuy/Build/Products/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App is not x86_64. Running lipo -info:
Non-fat file: /Users/dhavalkansara/Library/Developer/Xcode/DerivedData/Runner-dwtuugielmxutoatpwezjbbcemuy/Build/Products/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App is architecture: arm64
Command /bin/sh failed with exit code 1
```

**Comment by vijaykholiya** [comment]
I had the same issue for the framework Twilio Chat client. And when I went into carthage folder and inside it open the Info.plist file for Twilio Chat client framework the minimum deployment target was set to 11.0 but I was developing application for 10.0 so it was missing those architectures.

How I resolved: I downgraded my Twilio Chat Client framework to earlier versions where deployment target was set to 9.0 and boom it works.
